### PR TITLE
new(plugin): eclipse mosquitto mqtt

### PR DESCRIPTION
**Pull request by sdepassio**
## Description

New plugin to monitor Eclipse Mosquitto.

**Fixes** CTOR-374

### CODE REVIEW

File: src/apps/eclipse/mosquitto/mqtt/mode/clients.pm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self              = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+    });
```
Comment (Evan-Adam) [inline src/apps/eclipse/mosquitto/mqtt/mode/clients.pm:61]:
```suggestion

```

File: src/apps/eclipse/mosquitto/mqtt/mode/messages.pm [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self              = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+    });
```
Comment (Evan-Adam) [inline src/apps/eclipse/mosquitto/mqtt/mode/messages.pm:61]:
```suggestion

```

File: src/centreon/plugins/mqtt.pm [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    my $messages_received = 0;
+    while ($messages_received == 0 and Time::HiRes::time() < $endtime) {
+        $self->{mqtt}->tick(5);
+        $messages_received = scalar keys %mqtt_received;
+    }
+    eval {
+        $self->{mqtt}->unsubscribe($options{topic});
+    };
+    if (%mqtt_received) {
+        return %mqtt_received{$options{topic}};
```
[the remark this answers is not in the document]
Comment (sdepassio) [reply src/centreon/plugins/mqtt.pm:126]: A string is returned by "query" and a hash by "queries"

File: src/apps/eclipse/mosquitto/mqtt/mode/numericvalue.pm [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    }
+    if ($value !~ /^-?\d+(?:\.\d+)?$/) {
+        $self->{output}->output_add(
+            severity  => 'UNKNOWN',
+            short_msg => 'topic value is not numeric (' . $value . ')'
+        );
+        return;
+    }
+
+    if (!defined($value)) {
```
[the remark this answers is not in the document]
Comment (sdepassio) [reply src/apps/eclipse/mosquitto/mqtt/mode/numericvalue.pm:144]: the regex allows to extract a number from a string (for example if there is a unit with the value)

File: src/apps/eclipse/mosquitto/mqtt/mode/numericvalue.pm [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        $self->{output}->add_option_msg(short_msg => "Cannot find information");
```
Comment (omercier) [inline src/apps/eclipse/mosquitto/mqtt/mode/numericvalue.pm:145]: For troubleshooting purpose, I would have preferred that the message reads "Could not find a numeric value in '$value'" but the evaluated string from where we try to get it is lost at this point

**Review by omercier — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**